### PR TITLE
Clarify that some interface are not meant to be implemented

### PR DESCRIPTION
**Pull request by CarlSchwan**
This make sure that apps don't implement these interfaces and give us
more liberty to add new methods to them (as long as it is compatible
from a usage point of view). This just documents something we were already
doing before.

Signed-off-by: Carl Schwan <carl@carlschwan.eu>

### CODE REVIEW

File: lib/public/Share/IManager.php [hunk 1]
```diff
@@ -37,7 +37,11 @@
 use OCP\Share\Exceptions\ShareNotFound;
 
 /**
- * Interface IManager
+ * This interface allows to manage sharing files betweem users.
```
Comment (PVince81) [inline lib/public/Share/IManager.php:40]:
```suggestion
 * This interface allows to manage sharing files between users and groups.
```

**Review by PVince81 — APPROVED** [review]
:+1: as discussed, see suggestion

**Comment by PVince81** [comment]
here are two examples where the interface needed extending:
- https://github.com/nextcloud/server/pull/31194/files#diff-6c825986578290fe355bffe05c5232629a36e554d48f5d889bb2c1ec73645265R1790
- https://github.com/nextcloud/server/pull/31220/files#diff-6c825986578290fe355bffe05c5232629a36e554d48f5d889bb2c1ec73645265R968 @StCyr

**Review by ChristophWurst — APPROVED** [review]
We could even go with *must not*

**Review by juliusknorr — APPROVED** [review]

**Review by PVince81 — APPROVED** [review]
:+1:

**Comment by PVince81** [comment]
nice!

please make phpcs happy

File: lib/public/DB/IResult.php [hunk 2]
```diff
@@ -29,6 +29,18 @@
 use PDO;
 
 /**
+ * This interface represent the result of a database query.
```
Comment (MichaIng) [inline lib/public/DB/IResult.php:32]:
```suggestion
 * This interface represents the result of a database query.
```

**Review by MichaIng — APPROVED** [review]

**Pull request merged**